### DISCUSSION TITLE
How to set identity.issuer.crtExpiry if cert-manager rotates the issuer cert?

### DISCUSSION THREAD

**Discussion by sysarch-repo** [linkerd/linkerd2#6738] [Q&A]
Cert-Manager can be used to rotate the certificate of the issuer - see https://linkerd.io/2.10/tasks/automatically-rotating-control-plane-tls-credentials/

At the same time, identity.issuer.crtExpiry must be provided during linkerd install and the value must match the expiry date of the issuer cert. 

How does this align? The expectation would be making the identity.issuer.crtExpiry parameter optional. Thoughts?

**Comment by alpeb** [comment]
Assuming you're using Helm, the way to flag the templates that you'd like an external method to generate your issuer cert is to use `--set identity.issuer.scheme=kubernetes.io/tls`. In that case, `identity.issuer.crtExpiry` is not required; you can check that in the [issuer template's logic](https://github.com/linkerd/linkerd2/blob/36ce5538ebb9cf18b2b401f29c6a0afdebfbbf3f/charts/linkerd2/templates/identity.yaml#L6).

**Reply by sysarch-repo** [reply]
@alpeb, thanks for the prompt answer. Very appreciated!
